build_runner_core/lib/src/generate/build_definition.dart

```diff
@@ -374,26 +374,43 @@ class _Loader {
     return targets.asyncExpand(_listAssetIds).toSet();
   }
 
-  Stream<AssetId> _listAssetIds(TargetNode targetNode) async* {
-    for (final glob in targetNode.sourceIncludes) {
-      yield* _environment.reader
-          .findAssets(glob, package: targetNode.package.name)
-          .where((id) => !targetNode.excludesSource(id));
-    }
+  Stream<AssetId> _listAssetIds(TargetNode targetNode) {
+    var controller = new StreamController<AssetId>();
+    () async {
+      try {
+        for (final glob in targetNode.sourceIncludes) {
+          await controller.addStream(
```

Comment:
do we want to try to fire off multiple glob searches concurrently? I don't think we were before, but we could try it.

```dart
Stream<T> _mergeAll<T>(Iterable<Stream<T>> streams) => streams.first.transform(mergeAll(streams.skip(1)));

Stream<AssetId> _listAssetIds(TargetNode targetNode) {
  return _mergeAll(targetNode.sourceIncludes.map((glob) =>
      _listIdsSafe(glob, package: targetNode.package.name)
          .where((id) => !targetNode.excludesSource(id)));
}
```

Reply:
done